utils.bal

```diff
@@ -0,0 +1,361 @@
+            count = count + 1;
+        }
+        path = path + personFields[count];
+    }
+    return path;
+}
+
+# Prepare URL with read masks.
+# 
+# + url - Url to be appended
+# + readMasks - String array of fields to be fetched
+# + return - The prepared URL string
+isolated function prepareUrlWithReadMasks(string url, string[] readMasks) returns string {
+    string path = url;
+    int count = 0;
+    while (count < (readMasks.length() - 1)) {
+        path = path + readMasks[count] + ",";
+        count = count + 1;
+    }
+    path = path + readMasks[count];
+    return path;
+}
+
+# Prepare URL with optional sources.
+# 
+# + url - Url to be appended
+# + sources - An string array of sources to be restricted
+# + return - The prepared URL string
+isolated function prepareUrlWithOptionalSources(string url, string[]? sources) returns string {
+    string path = url;
+    if (sources is string[]) {
+        path = url + "&sources=";
+        int count = 0;
+        while (count < (sources.length() - 1)) {
+            path = path + sources[count] + ",";
+            count = count + 1;
+        }
+        path = path + sources[count];
+    }
+    return path;
+}
+
+# Prepare URL with copy masks.
+# 
+# + copyMasks - An string array of fields to be copied
+# + return - The prepared URL string
+isolated function prepareCopyMaskString(string[] copyMasks) returns string {
+    string path = "";
+    int count = 0;
+    while (count < (copyMasks.length() - 1)) {
+        path = path + copyMasks[count] + ",";
+        count = count + 1;
+    }
+    path = path + copyMasks[count];
+    return path;
+}
+
+# Prepare URL with read masks.
+# 
+# + readMasks - String array of fields to be fetched
+# + return - The prepared URL string
+isolated function prepareReadMaskString(string[] readMasks) returns string {
+    string path = "";
+    int count = 0;
+    while (count < (readMasks.length() - 1)) {
+        path = path + readMasks[count] + ",";
+        count = count + 1;
+    }
+    path = path + readMasks[count];
+    return path;
+}
+
+# Prepare URL with read group fields.
+# 
+# + readGroupFields - String array of fields to be fetched
+# + return - The prepared URL string
+isolated function prepareReadGroupFieldsString(string[] readGroupFields) returns string {
+    string path = "";
+    int count = 0;
+    while (count < (readGroupFields.length() - 1)) {
+        path = path + readGroupFields[count] + ",";
+        count = count + 1;
+    }
+    path = path + readGroupFields[count];
+    return path;
+}
+
+# Prepare URL with for batch operations.
+# 
+# + resourceNames - String array of resource names
+# + return - The prepared URL string
+isolated function prepareResourceString(string pathReceived, string[] resourceNames) returns string {
+    string path= pathReceived;
+    int count = 0;
+    while (count < (resourceNames.length() - 1)) {
+        path = path + CONTACTGROUP_PATH + resourceNames[count] + "&";
```

Comment:
Can use constants for symbols like "&".
Check all the places for similar issues.